spiel/spiel.h

```diff
@@ -1282,6 +1282,30 @@ struct m0_proc_counter {
 	struct m0_spiel_bcrec **pc_bcrec;
 };
 
+/**
+ * Counter stats allocated in motr, shall be freed in caller side
+ * 
+ * @param[in] count_stats   counter stats pointer for the rec and key
+ * @param[in] failed_index  counter stats need to free till 
+ *                          failed index value
+ */
+void m0_spiel_count_stats_free(struct m0_proc_counter *count_stats,
```

Comment:
I think this API can be removed, as fini is taking care of freeing the elements of the array. I don't see a usecase where anyone would want to free elements starting from 0 to failed_index, but not all the elements of the array.

Reply:
There is a instance inside m0_spiel_proc_counters_fetch() where it is trying to free until the failed index, so retaining this API.